File: PartnerApi/PHP/b2_reserve_trial_create_account.php
<?php

$api_url = ''; // Provided by b2_authorize_account
$auth_token = ''; // Provided by b2_authorize_account
$member_email = 'new-b2-reserve-member@mycompany.com'; // Email address for account being created
$term_in_days = 7; // Length of trial period in days
$storage_in_tb = 10; // Amount of storage in TB

$postData = json_encode([
    'email' => $member_email,
    'term' => $term_in_days,
    'storage' => $storage_in_tb,
]);

$ch = curl_init($api_url . '/b2api/v4/b2_reserve_trial_create_account');
curl_setopt_array($ch, [
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_POST           => true,
    CURLOPT_HTTPHEADER     => [
        "Authorization: $auth_token",
        'Content-Type: application/json',
    ],
    CURLOPT_POSTFIELDS     => $postData,
]);

echo curl_exec($ch), PHP_EOL;

File: PartnerApi/C#/b2_reserve_trial_create_account.cs
using System.Text;

class Program
{
    static readonly HttpClient Http = new();

    static async Task Main()
    {
        const string apiUrl = ""; // Provided by b2_authorize_account
        const string authToken = ""; // Provided by b2_authorize_account
        const string memberEmail = "new-b2-reserve-member@mycompany.com"; // Email address for account being created
        const int termInDays = 7; // Length of trial period in days
        const int storageInTB = 10; // Amount of storage in TB

        var postData = $@" {{ ""email"": ""{memberEmail}"", ""term"": {termInDays}, ""storage"": {storageInTB} }} ";

        Http.DefaultRequestHeaders.TryAddWithoutValidation("Authorization", authToken);

        var response = await Http.PostAsync(
            $"{apiUrl}/b2api/v4/b2_reserve_trial_create_account",
            new StringContent(postData, Encoding.UTF8, "application/json")
        );

        Console.WriteLine(await response.Content.ReadAsStringAsync());
    }
}